### ISSUE TITLE
Invalid XPath expression in expand.xsl

### ISSUE THREAD

**Issue by Victor-Dev-Init** [schxslt/schxslt#142]
Within a KoSIT project called xml-mutate (A declarative XML instance Mutating and Testing tool) we are investigating the added value of schematron processor SchXslt. Till now we availed of a self-developed schematron compiler that uses the Skeleton-Implementation for XSLT2 and the Saxon-Framework (http://schematron.com/front-page/the-schematron-skeleton-implementation).

Due to the fact that the Schematron/schematron project will be archived and read-only from 1 October 2020 we decided to look into this SchXslt schematron processor as advised by the Github community.

Unfortuntely our initial tests with the CLI project led to the following error (please go on reading after error):

```
Error evaluating (fn:replace(...)) in xsl:variable/@select on line 81 column 132 of expand.xsl:
  FORX0004: Invalid replacement string in replace(): \ character must be followed by \ or $
at function schxslt:replace-params on line 72 of expand.xsl:
     invoked by function call at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#82
at function schxslt:replace-params on line 72 of expand.xsl:
     invoked by function call at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#82
at function schxslt:replace-params on line 72 of expand.xsl:
     invoked by function call at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#82
at function schxslt:replace-params on line 72 of expand.xsl:
     invoked by function call at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#82
at function schxslt:replace-params on line 72 of expand.xsl:
     invoked by function call at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#82
at function schxslt:replace-params on line 72 of expand.xsl:
     invoked by function call at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#82
at function schxslt:replace-params on line 72 of expand.xsl:
     invoked by function call at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#82
at function schxslt:replace-params on line 72 of expand.xsl:
     invoked by function call at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#82
at function schxslt:replace-params on line 72 of expand.xsl:
     invoked by function call at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#82
at function schxslt:replace-params on line 72 of expand.xsl:
     invoked by function call at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#82
at function schxslt:replace-params on line 72 of expand.xsl:
     invoked by function call at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#82
at function schxslt:replace-params on line 72 of expand.xsl:
     invoked by function call at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#82
at function schxslt:replace-params on line 72 of expand.xsl:
     invoked by function call at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#82
at function schxslt:replace-params on line 72 of expand.xsl:
     invoked by function call at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#82
at function schxslt:replace-params on line 72 of expand.xsl:
     invoked by function call at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#82
at function schxslt:replace-params on line 72 of expand.xsl:
     invoked by function call at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#82
at function schxslt:replace-params on line 72 of expand.xsl:
     invoked by function call at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#68
  In template rule with match="element(Q{http://purl.oclc.org/dsdl/schematron}rule)/attribute(Q{}context)" on line 66 of expand.xsl
     invoked by xsl:apply-templates at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#31
  In template rule with match="( element() | text() | comment() | processing-instruction() )" on line 29 of expand.xsl
     invoked by xsl:apply-templates at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#59
  In template rule with match="element(Q{http://purl.oclc.org/dsdl/schematron}pattern)[@is-a]" on line 54 of expand.xsl
     invoked by xsl:apply-templates at jar:file:/C:/Users/vdelcampocam/Downloads/schxslt-master/cli/target/schxslt-cli.jar!/xslt/2.0/expand.xsl#22
Exception in thread "main" name.dmaus.schxslt.SchematronException: Error compiling Schematron to transformation stylesheet
        at name.dmaus.schxslt.Schematron.compile(Schematron.java:264)
        at name.dmaus.schxslt.Schematron.getValidationStylesheet(Schematron.java:205)
        at name.dmaus.schxslt.Schematron.validate(Schematron.java:178)
        at name.dmaus.schxslt.Schematron.validate(Schematron.java:163)
        at name.dmaus.schxslt.cli.Application.execute(Application.java:106)
        at name.dmaus.schxslt.cli.Application.main(Application.java:66)
Caused by: net.sf.saxon.trans.XPathException: Invalid replacement string in replace(): \ character must be followed by \ or $
        at net.sf.saxon.functions.Replace.call(Replace.java:77)
        at net.sf.saxon.functions.Replace.call(Replace.java:24)
        at net.sf.saxon.expr.FunctionCall.iterate(FunctionCall.java:532)
        at net.sf.saxon.expr.Expression.evaluateItem(Expression.java:851)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.UserFunctionCall.evaluateArguments(UserFunctionCall.java:640)
        at net.sf.saxon.expr.UserFunctionCall.evaluateArguments(UserFunctionCall.java:622)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:516)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:80)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:76)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:392)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:30)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:354)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:28)
        at net.sf.saxon.expr.instruct.Choose.evaluateItem(Choose.java:947)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:601)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:80)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:76)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:392)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:30)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:354)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:28)
        at net.sf.saxon.expr.instruct.Choose.evaluateItem(Choose.java:947)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:601)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:80)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:76)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:392)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:30)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:354)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:28)
        at net.sf.saxon.expr.instruct.Choose.evaluateItem(Choose.java:947)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:601)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:80)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:76)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:392)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:30)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:354)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:28)
        at net.sf.saxon.expr.instruct.Choose.evaluateItem(Choose.java:947)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:601)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:80)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:76)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:392)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:30)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:354)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:28)
        at net.sf.saxon.expr.instruct.Choose.evaluateItem(Choose.java:947)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:601)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:80)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:76)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:392)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:30)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:354)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:28)
        at net.sf.saxon.expr.instruct.Choose.evaluateItem(Choose.java:947)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:601)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:80)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:76)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:392)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:30)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:354)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:28)
        at net.sf.saxon.expr.instruct.Choose.evaluateItem(Choose.java:947)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:601)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:80)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:76)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:392)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:30)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:354)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:28)
        at net.sf.saxon.expr.instruct.Choose.evaluateItem(Choose.java:947)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:601)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:80)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:76)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:392)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:30)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:354)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:28)
        at net.sf.saxon.expr.instruct.Choose.evaluateItem(Choose.java:947)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:601)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:80)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:76)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:392)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:30)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:354)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:28)
        at net.sf.saxon.expr.instruct.Choose.evaluateItem(Choose.java:947)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:601)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:80)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:76)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:392)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:30)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:354)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:28)
        at net.sf.saxon.expr.instruct.Choose.evaluateItem(Choose.java:947)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:601)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:80)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:76)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:392)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:30)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:354)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:28)
        at net.sf.saxon.expr.instruct.Choose.evaluateItem(Choose.java:947)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:601)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:80)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:76)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:392)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:30)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:354)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:28)
        at net.sf.saxon.expr.instruct.Choose.evaluateItem(Choose.java:947)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:601)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:80)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:76)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:392)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:30)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:354)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:28)
        at net.sf.saxon.expr.instruct.Choose.evaluateItem(Choose.java:947)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:601)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:80)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:76)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:392)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:30)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:354)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:28)
        at net.sf.saxon.expr.instruct.Choose.evaluateItem(Choose.java:947)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:601)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:80)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:76)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:392)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:30)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:354)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:28)
        at net.sf.saxon.expr.instruct.Choose.evaluateItem(Choose.java:947)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:601)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:80)
        at net.sf.saxon.functions.ScalarSystemFunction$1.evaluateItem(ScalarSystemFunction.java:76)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:392)
        at net.sf.saxon.expr.CastExpression.evaluateItem(CastExpression.java:30)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:354)
        at net.sf.saxon.expr.AtomicSequenceConverter.evaluateItem(AtomicSequenceConverter.java:28)
        at net.sf.saxon.expr.instruct.Choose.evaluateItem(Choose.java:947)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:191)
        at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:188)
        at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:601)
        at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541)
        at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477)
        at net.sf.saxon.expr.Expression.evaluateAsString(Expression.java:916)
        at net.sf.saxon.expr.instruct.SimpleNodeConstructor.processLeavingTail(SimpleNodeConstructor.java:218)
        at net.sf.saxon.expr.instruct.Block.processLeavingTail(Block.java:735)
        at net.sf.saxon.expr.instruct.TemplateRule.applyLeavingTail(TemplateRule.java:352)
        at net.sf.saxon.trans.Mode.applyTemplates(Mode.java:533)
        at net.sf.saxon.expr.instruct.ApplyTemplates.apply(ApplyTemplates.java:300)
        at net.sf.saxon.expr.instruct.ApplyTemplates.process(ApplyTemplates.java:251)
        at net.sf.saxon.expr.instruct.ElementCreator.processLeavingTail(ElementCreator.java:352)
        at net.sf.saxon.expr.instruct.Copy.processLeavingTail(Copy.java:431)
        at net.sf.saxon.expr.instruct.TemplateRule.applyLeavingTail(TemplateRule.java:352)
        at net.sf.saxon.trans.Mode.applyTemplates(Mode.java:533)
        at net.sf.saxon.expr.instruct.ApplyTemplates.apply(ApplyTemplates.java:300)
        at net.sf.saxon.expr.instruct.ApplyTemplates.processLeavingTail(ApplyTemplates.java:255)
        at net.sf.saxon.expr.instruct.Block.processLeavingTail(Block.java:735)
        at net.sf.saxon.expr.instruct.Instruction.process(Instruction.java:132)
        at net.sf.saxon.expr.instruct.ElementCreator.processLeavingTail(ElementCreator.java:352)
        at net.sf.saxon.expr.instruct.Copy.processLeavingTail(Copy.java:431)
        at net.sf.saxon.expr.LetExpression.processLeavingTail(LetExpression.java:723)
        at net.sf.saxon.expr.instruct.Block.processLeavingTail(Block.java:735)
        at net.sf.saxon.expr.instruct.TemplateRule.applyLeavingTail(TemplateRule.java:352)
        at net.sf.saxon.trans.Mode.applyTemplates(Mode.java:533)
        at net.sf.saxon.expr.instruct.ApplyTemplates.apply(ApplyTemplates.java:300)
        at net.sf.saxon.expr.instruct.ApplyTemplates.processLeavingTail(ApplyTemplates.java:255)
        at net.sf.saxon.expr.instruct.Block.processLeavingTail(Block.java:735)
        at net.sf.saxon.expr.instruct.Instruction.process(Instruction.java:132)
        at net.sf.saxon.expr.instruct.ElementCreator.processLeavingTail(ElementCreator.java:352)
        at net.sf.saxon.expr.instruct.Copy.processLeavingTail(Copy.java:431)
        at net.sf.saxon.expr.instruct.Block.processLeavingTail(Block.java:735)
        at net.sf.saxon.expr.instruct.NamedTemplate.expand(NamedTemplate.java:243)
        at net.sf.saxon.expr.instruct.CallTemplate$CallTemplatePackage.processLeavingTail(CallTemplate.java:519)
        at net.sf.saxon.trans.Mode.applyTemplates(Mode.java:451)
        at net.sf.saxon.trans.rules.TextOnlyCopyRuleSet.process(TextOnlyCopyRuleSet.java:71)
        at net.sf.saxon.trans.Mode.applyTemplates(Mode.java:495)
        at net.sf.saxon.trans.XsltController.applyTemplates(XsltController.java:746)
        at net.sf.saxon.s9api.AbstractXsltTransformer.applyTemplatesToSource(AbstractXsltTransformer.java:347)
        at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:349)
        at net.sf.saxon.jaxp.TransformerImpl.transform(TransformerImpl.java:71)
        at name.dmaus.schxslt.Schematron.applyPipeline(Schematron.java:275)
        at name.dmaus.schxslt.Schematron.compile(Schematron.java:257)
        ... 5 more

```

We tested with a public XRechnung schematron file, which we have been using in our own schematron compiler. We are running the following command for CLI:

```
java -jar schxslt-cli.jar -s C:\Users\xxx\Downloads\schxslt-master\cli\src\main\resources\XRechnung-UBL-validation-Invoice.sch
```

Here are the schematron files for this command run: 
[XRechnung-UBL-validation-Invoice.zip](https://github.com/schxslt/schxslt/files/5207550/XRechnung-UBL-validation-Invoice.zip)

We also tested with a "dummy" schematron files: 
[book.zip](https://github.com/schxslt/schxslt/files/5207557/book.zip)

... by running the following CLI command:
```
java -jar schxslt-cli.jar -s C:\Users\xxx\Downloads\schxslt-master\cli\src\main\resources\book.sch
```
But nothing happend, no error, no success, nothing at all

FYI: Initial tests with the java project are not looking any better.

**Comment by dmj** [comment]
I can confirm this bug

**Comment by dmj** [comment]
The bug is triggered by \ characters in parameter values. Related to #60.

**Comment by dmj** [comment]
This should be fixed in master.

**Comment by Victor-Dev-Init** [comment]
It does not seem to be fixed. There is no error now, but nothing at all happens. No error, no logs, nothing

**Comment by dmj** [comment]
If I create a CLI from master by running ```mvn package``` and validate the following file

```
<book/>
```
with the Schematron from ```books.zip``` I get:

```
java -jar ~/projects/schxslt/cli/target/schxslt-cli.jar  -v -s book.sch -d book.xml
[invalid] /home/dmaus/downloads/bug/book.xml
[invalid] /home/dmaus/downloads/bug/book.xml failed-assert /Q{}book[1] [Book-1] A book must always have a publisher.
[invalid] /home/dmaus/downloads/bug/book.xml failed-assert /Q{}book[1] [Book-2] A book must always have a number of chapters.
```

**Comment by Victor-Dev-Init** [comment]
Oh I see now David. With the CLI you can only validate an xml against an input schematron file right?

You can't compile the input schematron file (sch) into a xslt?

But compiling the sch file (and not validating an xml against this sch) should be possible with the java project?

**Comment by dmj** [comment]
That's right: The CLI performs validation. The Java project does not expose the compilation yet but I will prepare a 2.1 release of SchXslt java to address this issue (https://github.com/schxslt/schxslt-java/issues/34).

**Comment by dmj** [comment]
Sorry, my fault: The Java class Schematron name.dmaus.schxslt.Schematron exposes a function getValidationStylesheet() that compiles and returns the validation stylesheet.

**Comment by dmj** [comment]
By default SchXslt Java uses whatever XSL transformation factory is setup by the ```javax.xml.transform.TransformerFactory``` factory. If you want to use Schematron with XSLT 2 you need to supply the required TransformerFactory, i.e. the Saxon implementation. You can do this by either setting the afformentioned propery or set the TransformerFactory with a call to ```Schematron.withTransformerFactory```.

**Comment by Victor-Dev-Init** [comment]
I am getting now the same error I had with the CLI project:
```
Caused by: net.sf.saxon.trans.XPathException: Invalid replacement string in replace(): \ character must be followed by \ or $
```

But this is because the latest released version of the java project (2.0.1) is referring to 1.4.2 of name.dmaus.schxslt:schxslt and this version has still the old expand.xsl

**Comment by dmj** [comment]
I can confirm. I published version 2.0.2 of SchXslt that requires SchXslt 1.4.7 or higher. It should be available at Maven Central soon.
